### PR TITLE
#54 - Added integration test for pushing image that already exists

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #54 
Added integration test case for pushing an image that is already in the repository

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#98](https://github.com/artipie/docker-adapter/issues/98) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/1b20b00c-c145-4032-a3ea-5eb519d1b1ec); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/docker/http/DockerSliceITCase.java [hunk 1]
```diff
@@ -143,6 +143,33 @@ void shouldPush() throws Exception {
...
+            "%s:%s",
+            "sha256",
+            "cb8a924afdf0229ef7515d9e5b3024e23b3eb03ddbba287f4a19c6ac90b8d221"
+        );
+        final String original = String.format("alpine@%s", digest);
+        this.run("pull", original);
+        final String remote = String.format("%s/my-alpine", this.repo);
+        this.run("tag", original, remote);
+        final String output = this.run("push", remote);
+        MatcherAssert.assertThat(
```
Comment (Vatavuk) [inline src/test/java/com/artipie/docker/http/DockerSliceITCase.java:159]: @olegmoz we can simplify to:
```
MatcherAssert.assertThat(
  output,
  Matchers.allOf(
            new StringContains(false, "..."),
            new StringContains(false, "...")
  )
);

**Comment by olegmoz** [comment]
@g4s8 no reviewer assigned for 22hrs, please look into this

**Comment by Vatavuk** [comment]
@g4s8 I'm assigned, @olegmoz I left you a comment

**Comment by olegmoz** [comment]
@Vatavuk oops, my bad. Thank you for review, made the changes

**Comment by 0crat** [comment]
Job was finished in 23 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable  ([here](https://github.com/artipie/docker-adapter/issues/98#issuecomment-621806035))

@sereshqua There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@3b2f8302-e30a-4cce-a229-5dee4cefb2b5, thread: PQ-CT2E6TK9B
com.zerocracy.farm.reactive.StkRuntime[76] java.lang.IllegalStateException: org.codehaus.groovy.runtime.InvokerInvocationException in com.zerocracy.stk.pm.cost.pay_cash
org.codehaus.groovy.reflection.CachedMethod[105] org.codehaus.groovy.runtime.InvokerInvocationException: java.lang.OutOfMemoryError: unable to create new native thread
java.lang.Thread[-2] java.lang.OutOfMemoryError: unable to create new native thread
```

[0.53.13](https://github.com/zerocracy/farm): CID: [e2db21a4-7702-4a9d-93bd-34efa32c08f9](https://www.0crat.com/footprint/CT2E6TK9B/e2db21a4-7702-4a9d-93bd-34efa32c08f9), Type: "Pay cash"

**Pull request merged**